Changelog July 4 (#500)
CloudCannon/platform-documentation: changelogs/2025-07-04_updated-file-and-url-inputs.mdx

---
_schema: default
title: Updated File and URL inputs
---
This release updated the appearance and functionality to File and URL Inputs, improved copy in the Editing Interface *Context menu*, and improved error messaging for image previews on File Cards.

It also addressed an issue affecting Context Boxes on Boolean Inputs.

## Features & Improvements

### File Inputs

<comp.DocsImage path="https://cc-dam.imgix.net/documentation/images/change-log/2025-07/CloudCannon-Changelog-File-Input.png" alt="A screenshot of the File Input in the Data Editor shows the preview and metadata for the selected CSS file." title="The File Input" type="ui-snippet"/>

* File Inputs now have a text field, *Clear* button, and *Select file* button at the top of the drag-and-drop area.
* The *Select file* button opens a dropdown with the *Upload a new file* and *Select existing file* options.

> * When you select a file, CloudCannon will display metadata about that file if it is a document, or a preview if it is an image.

Reviewer: ```suggestion
* When you select a file, CloudCannon will display a preview if it is an image, or metadata about the file otherwise.
```
